tools/influx_si/influx_si.xml

```diff
@@ -1,232 +1,63 @@
-<tool id="influx_si" name="influx_si" version="@TOOL_VERSION@+galaxy1">
-   <description>metabolic flux analysis</description>
-   <macros>
-        <token name="@TOOL_VERSION@">5.1.0</token>
-   </macros>
-   <requirements>
-      <requirement type="package" version="@TOOL_VERSION@">influx_si</requirement>
-   </requirements>
-   <version_command>influx_s --version</version_command>
-   <command detect_errors="exit_code"><![CDATA[
-       #set $inp_all = [$input_main, ($prlexp.input_aux if $prlexp.input_aux else [])]
-       #if $si.s_i == 'i':
-         #silent $inp_all.append($si.input_ti)
-       #end if
-
-       mkdir outdir && cd outdir &&
-
-       #for $inp in $inp_all:
-         #for $i in $inp:
-           ln -s '$i' '$i.element_identifier' &&
-         #end for
+<tool id="influx_si" name="influx_si" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="21.09">
+    <description>
+        Estimate metabolic fluxes and concentrations by fitting simulated labeling in metabolites to NMR/MS measurements
+    </description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">influx_si</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[  
+      
+       mkdir outdir;
+       #for $inp in $input_main:
+        unzip $inp;
        #end for
-
+              
        influx_$si.s_i
-            $opt.noopt
-            $opt.noscale
-            #if $opt.meth != 'None':
-               $opt.meth
-            #end if
-            $opt.fullsys
-            $opt.emu
-            $opt.irand
-            #if $opt.sens:
-             --sens='$opt.sens'
-            #end if
-            #if $opt.cupx:
-             --cupx='$opt.cupx'
-            #end if
-            #if $opt.cupn:
-             --cupn='$opt.cupn'
-            #end if
-            #if $opt.cupp:
-             --cupp='$opt.cupp'
-            #end if
-            #if $opt.clownr:
-             --clownr='$opt.clownr'
-            #end if
-            #if $opt.cinout:
-             --cinout='$opt.cinout'
-            #end if
-            #if $opt.clowp:
-             --clowp='$opt.clowp'
-            #end if
-            #if $opt.np:
-             --np='$opt.np'
-            #end if
-            $opt.ln
-            $opt.sln
-            $opt.tikhreg
-            $opt.lim
-            #if $opt.zc:
-             --zc='$opt.zc'
-            #end if
-            $opt.ffguess
-            #if $opt.iseries:
-             --iseries='$opt.iseries'
-            #end if
-            #if $opt.seed:
-             --seed='$opt.seed'
-            #end if
-            #if $opt.excl_outliers:
-             --excl_outliers '$opt.excl_outliers'
-            #end if
-            $opt.nocalc
-            $opt.addnoise
-            $opt.TIMEIT
-            #if $opt.tblimit:
-             --tblimit='$opt.tblimit'
-            #end if
-
+         @COMMAND_OPT@
```

Comment:
@bgruening @bernt-matthias is it a good practice to transfert inputs stuffs (command template and <INPUTS/>) in a macro ?

It's less readable I find.

Reply:
Yes, it less readable, but if we actually reuse it multiple times it might pay off because you don't repeat yourself and need to fix bugs in multiple places. We do not have a strict rule here.

Reply:
100%

Reply:
In hindsight, I should have agreed with @lecorguille and @bgruening . Here the use of macros seems to be not a good idea. It does not avoid any code duplication (as far as I can see) and just makes the code difficult to read. 

And in particular it makes it difficult to understand the diff (ie the change made here). Seems that now zip files are the input. Should be trivial to move the input section and the command part back, or?